llvm/lib/Analysis/ConstantFolding.cpp

```diff
@@ -151,8 +151,14 @@ Constant *FoldBitCast(Constant *C, Type *DestTy, const DataLayout &DL) {
     return FoldBitCast(ConstantVector::get(Ops), DestTy, DL);
   }
 
+  // Some of what follows may extend to cover scalable vectors but the current
+  // implementation is fixed length specific.
+  if (!isa<FixedVectorType>(C->getType()))
+    return ConstantExpr::getBitCast(C, DestTy);
+
   // If this is a bitcast from constant vector -> vector, fold it.
-  if (!isa<ConstantDataVector>(C) && !isa<ConstantVector>(C))
+  if (!isa<ConstantDataVector>(C) && !isa<ConstantVector>(C) &&
+      !isa<ConstantInt>(C) && !isa<ConstantFP>(C))
```

Comment:
Refers to line 206: Do we need to add ConstantInt and ConstantFP checks here too?

Reply:
I was going to say no because it should be safe to not perform anymore transformations.  However, if I'm reading it correctly I think that block of code is just broken upstream?

For the case where constant folding does not happen it is effectively returning `ConstantExpr::getBitCast(C, SrcAsIntTy)` and by this point we already know `SrcAsIntTy != DestVTy`?